src/components/data-management/AnalysisModal.jsx

```diff
@@ -10,120 +10,112 @@ import {
   Col,
   List,
 } from 'antd';
-import { ClipLoader } from 'react-spinners';
+import moment from 'moment';
+import { useRouter } from 'next/router';
 import EditableField from '../EditableField';
-import { updateExperiment, saveExperiment } from '../../redux/actions/experiments';
-import validateInputs, { rules } from '../../utils/validateInputs';
+import { updateExperiment } from '../../redux/actions/experiments';
+import {
+  updateProject,
+} from '../../redux/actions/projects';
+
+import { runGem2s } from '../../redux/actions/pipeline';
 
 const { Title } = Typography;
 
-const NewExperimentModal = (props) => {
+const AnalysisModal = (props) => {
   const {
     visible,
     onLaunch,
     onCancel,
-    activeProject,
-    experiments,
   } = props;
 
   const dispatch = useDispatch();
+  const router = useRouter();
 
-  const [experimentsList, setExperimentsList] = useState([]);
-  const [numFieldsEditing, setNumFieldsEditing] = useState(0);
-  const [isWorking, setIsWorking] = useState(false);
+  const experiments = useSelector((state) => state.experiments);
+  const { activeProjectUuid } = useSelector((state) => state.projects.meta) || false;
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
 
-  useEffect(() => {
-    if (!activeProject?.experiments?.length > 0) return;
+  const getExperimentsList = () => activeProject.experiments
+    .map((experimentId) => experiments[experimentId])
+    .filter((experiment) => experiment !== undefined);
 
-    const updatedList = activeProject.experiments
-      .map((experimentId) => experiments[experimentId])
-      .filter((experiment) => experiment !== undefined);
+  const [experimentsList, setExperimentsList] = useState(getExperimentsList());
+  const [numFieldsEditing, setNumFieldsEditing] = useState(0);
 
+  useEffect(() => {
+    const updatedList = getExperimentsList();
     setExperimentsList(updatedList);
   }, [activeProject, experiments]);
-  useEffect(() => {
-    setIsWorking(!visible);
-  }, [visible]);
-
-  const validationChecks = [
-    rules.MIN_1_CHAR,
-    rules.ALPHANUM_DASH_SPACE,
-  ];
-
-  const renderAnalysisList = () => {
-    if (!experimentsList?.length > 0) {
-      return (
-        <Row justify='center'>
-          <ClipLoader size={30} color='#8f0b10' />
-        </Row>
-      );
-    }
```

Comment:
We don't need this code -- AnalysisModal will never get rendered if there is no experiment data. Currently in master, it gets rendered all the time, and when we shouldn't see it, it was made to be invisible